solutions/usaco-416.mdx

```diff
@@ -12,7 +12,81 @@ author: Brad Ma
 **Time Complexity** $\mathcal{O}(K \cdot N ^ 2)$
 
 <LanguageSection>
-  
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <fstream>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+int main() {
+	std::ifstream read("lazy.in");
+
+	int n, k;
+	read >> n >> k;
+
+	/*
+	 * the 45 degree rotation expands the size of the field
+	 * it also includes the cells with half distance
+	 */
+	vector<vector<int>> field(2 * n + 1, vector<int>(2 * n + 1));
+
+	// read input and store it rotated by 45 degrees
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			read >> field[i + j][n - i + j - 1];
+		}
+	}
+
+	/*
+	 * used to avoid non-lattice points &
+	 * ensures that the dummy 0's don't affect anything
+	 */
+	int t = (n + 1) % 2;
+
+	n = n * 2 - 1;
```

Comment:
i dont recommend changing the values for `n` because someone can miss it and be like wtf thats not what the `n` in the problem statement means!!

Reply:
yeah, i should probably have done that